runtime/interpreter/interpreter.go

```diff
@@ -978,6 +980,47 @@ func (interpreter *Interpreter) declareAttachmentValue(
 	return interpreter.declareCompositeValue(declaration, lexicalScope)
 }
 
+// evaluates all the implicit default arguments to the default destroy event
+//
+// the handling of default arguments makes a number of assumptions to simplify the implementation;
+// namely that a) all default arguments are lazily evaluated at the site of the invocation,
+// b) that either all the parameters or none of the parameters of a function have default arguments,
+// and c) functions cannot currently be explicitly invoked if they have default arguments
+//
+// if we plan to generalize this further, we will need to relax those assumptions
+func (interpreter *Interpreter) evaluateDefaultDestroyEvent(
+	containerComposite *CompositeValue,
+	compositeDecl *ast.CompositeDeclaration,
+	compositeType *sema.CompositeType,
+	locationRange LocationRange,
+) (arguments []Value) {
+	parameters := compositeDecl.DeclarationMembers().Initializers()[0].FunctionDeclaration.ParameterList.Parameters
+
+	interpreter.activations.PushNewWithParent(interpreter.activations.CurrentOrNew())
+	defer interpreter.activations.Pop()
+
+	var self MemberAccessibleValue = containerComposite
+	if containerComposite.Kind == common.CompositeKindAttachment {
+		var base *EphemeralReferenceValue
+		base, self = attachmentBaseAndSelfValues(interpreter, containerComposite)
+		interpreter.declareVariable(sema.BaseIdentifier, base)
+	}
+	interpreter.declareVariable(sema.SelfIdentifier, self)
+
+	for _, parameter := range parameters {
+		// lazily evaluate the default argument expressions
+		// note that we must evaluate them in the interpreter context that existed when the event
+		// was defined (i.e. the contract defining the resource) rather than the interpreter context
+		// that exists when the resource is destroyed. We accomplish this by using the original interpreter of the
+		// composite declaration, rather than the interpreter of the destroy expression
```

Comment:
Where is the "right" interpreter (even definition instead of destroy expression) used?

Maybe just use (or perform the same logic as) `Interpreter.invokeInterpretedFunction`, i.e. push the function's (event constructors) activation above.